tests/middleware/test_base.py

```diff
@@ -84,3 +84,46 @@ def homepage(request):
     response = client.get("/homepage")
     assert response.text == "Homepage"
     assert response.headers["Custom"] == "Example"
+
+
+def test_state_data_across_multiple_middlewares():
+    expected_value1 = "foo"
+    expected_value2 = "bar"
+
+    class aMiddleware(BaseHTTPMiddleware):
+        async def dispatch(self, request, call_next):
+            request.state.foo = expected_value1
+            print("a", dict(request))
```

Comment:
Don't you want to remove these print statements?

Reply:
Yes. Yes, I do. Thanks!